### PR TITLE
Bugfix: use virt operator image if provided

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
Small bug fix - use virt-operator custom image if provided.
This is another tiny follow-up to https://github.com/kubevirt/kubevirt/pull/8673, https://github.com/kubevirt/kubevirt/pull/8792 and https://github.com/kubevirt/kubevirt/pull/8811.

**Release note**:

```release-note
Bugfix: use virt operator image if provided
```

### CODE REVIEW

File: pkg/virt-operator/kubevirt.go [hunk 1]
```diff
@@ -754,6 +757,10 @@ func (c *KubeVirtController) generateInstallStrategyJob(config *operatorutil.Kub
 								"--dump-install-strategy",
 							},
 							Env: []k8sv1.EnvVar{
+								{
+									Name:  util.VirtOperatorImageEnvName,
+									Value: operatorImage,
+								},
 								{
 									// Deprecated, keep it for backwards compatibility
 									Name:  util.OldOperatorImageEnvName,
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-operator/kubevirt.go:766]: Perhaps I have been too cautious... IMO when we deprecate something we need to keep it for a reasonable time, and then we can delete it. Someone, externally, can could consume it. Don't you think?

[the remark this answers is not in the document]
Comment (jean-edouard) [reply pkg/virt-operator/kubevirt.go:766]: Someone could consume a virt-operator environment variable? I don't think anything outside of virt-operator itself can even see those, right?
Anyway, that can be addressed in a separate PR. What's done here is fine.
/approve

[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-operator/kubevirt.go:766]: I mean something like `kv get deploy virt-operator -o jsonpath="{.spec.template.spec.containers[0].env[?(@.name=='OPERATOR_IMAGE')].value}"`

**Comment by iholder101** [comment]
> /hold operator test lanes are failing with the same failing error @iholder101 can you have a look?

Sure, thanks for paying attention.
They're now fixed. I ended up removing the deprecated env variables from virt-operator deployment.

/hold cancel

**Pull request merged**